src/components/query/QueryDirective.js

```diff
@@ -352,8 +353,16 @@
 
     // Watcher/listener
     $scope.layers = $scope.map.getLayers().getArray();
-    $scope.layerFilter = gaLayerFilters.selectByRectangle;
-    $scope.$watchCollection('layers | filter:layerFilter', function(layers) {
+    $scope.selectByRectangleFilter = gaLayerFilters.selectByRectangle;
+    $scope.$watchCollection('layers | filter:selectByRectangleFilter',
```

Comment:
Since we use the same service for bbox and filter requests (features identify which use the database). I think the searchable property is the property to use. We can get rid of selectByRectangle stuff here.

But maybe I miss something.